src/Nexus.ts

```diff
@@ -208,6 +235,13 @@ class Nexus {
     return res;
   }
 
+  public static async createWithOAuth(credentials: types.IOAuthCredentials, config: types.IOAuthConfig, appName: string, appVersion: string, defaultGame: string, timeout?: number): Promise<Nexus> {
+    const res = new Nexus(appName, appVersion, defaultGame, timeout);
+    res.setOAuthCredentials(credentials);
```

Comment:
It looks kind of weird that oAuthConfig gets assigned directly but the credentials you have to call a function.
I think it would be nicer if you turned "setOAuthCredentials" into a setter, like this:
`private set oAuthCredentials(value: type.IOAuthCredentials)`

Reply:
I used the `setKey` / `setGame` as a guide here. The reason why there's a function for the credentials and not the config is because the config just sets the property. The credentials setter does other logic (setting the headers etc).

I'll make it a setter.

If the `oAuthCredentials` property is being renamed to `mOAuthCredentials` would you name the setter:
`private set oAuthCredentials(...)` or `private set mOAuthCredentials`?

Reply:
Good point re setKey/setGame, I only really noticed it because the assignment to oAuthCredentials and autchConfig were side-by-side.
tbh I'd be ok if you leave it as is, it's completely private anyway.

Re the setter: I'd call it private set oAuthCredentials(...)